jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1121,6 +1133,22 @@ typedef struct
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+
+/**
+ * Internal properties of TypedArray Object
+ */
+typedef struct
+{
+  /** The arraybuffer object behind the typedarray */
+  ecma_object_t *viewed_arraybuffer_p;
```

Comment:
Hm, I have an idea which could solve two issues:

Add a byte to extended object, which is
- 0 for non-typed-arrays
- log2(item_length)+1 for typed-arrays

This would:
- heavily simplify ecma_is_typedarray
- length could be computed as (buffer-length) >> (this byte-1)